### PR TITLE
Add "I don't know" option to Selective Service

### LINKED ISSUE #1320: Applicants should be able to select "I don't know" for Selective Service Record
As an applicant
I want to be able to select "I don't know" when asked about my Selective Service Record
So that I can give the most accurate information for my SSS status

Given I am on the Selective Service Record subsection
And I have answered "Yes" to being a male born after December 31st, 1959
Then I should see a question asking me "Have you registered with the Selective Service System (SSS)?"
And I should be able to choose "I don't know"

Given I have chosen "I don't know"
Then I should see a multiline text field asking me "Provide explanation"

[Link to SF-85 content guide for this section](https://docs.google.com/presentation/d/1LCahQUxtp2lcIjjVguzivokDe-cXEcZhVWzk-6S9pc0/edit#slide=id.g44d6d9eb88_2_82)

### PR DESCRIPTION
**Pull request by hursey013**
Fixes #1320 

Uses the `NotApplicable` component to add an "I don't  know" option to Selective Service record.

Includes a new database migration, so make sure to restart the server when testing.

screen shot 2019-01-17 at 10 47 34 am

### CODE REVIEW

File: api/templates/military-selective.xml [hunk 1]
```diff
@@ -9,6 +9,12 @@
     <LegalExemptionExplanation>{{textarea .props.Explanation}}</LegalExemptionExplanation>
     <LegalExemptionExplanationComment></LegalExemptionExplanationComment>
     <NumberComment></NumberComment>
+    <DNKExplanation></DNKExplanation>
```
Comment (gidjin) [inline api/templates/military-selective.xml]: Did you mean to add lines 12 and 13? Looks like duplicates of 18 and 19. If so do we need to fill in the explanation?

**Comment by gidjin** [comment]
Looks like this works. Should we show the _Provide an explanation_ box if the applicant selects _I don't know_? Also please see the commend about the api template xml, looks like there maybe some inadvertent duplication.

File: api/templates/military-selective.xml (hunk 1 above)
Comment (hursey013) [reply api/templates/military-selective.xml]: Admittedly, this is where my lack of knowledge on how the XML is expected to look downstream comes into play. It looks like in a lot of areas we have blank XML tags in the template, like `<LegalExemptionExplanationComment></LegalExemptionExplanationComment>`, so for consistency's sake I return the empty tags in both conditions, but I'm not totally sure that is necessary. The way I have it now returns the XML exactly as it was before, the only difference is in one case the explanation is in the `LegalExemptionExplanation` tag, and in the other the `DNKExplanation`. Open to change this around if you think we can omit the empty tags.

**Comment by hursey013** [comment]
@gidjin yes the explanation box should appear when you select `I don't know` - was it not appearing during your testing?

**Comment by gidjin** [comment]
It appeared as expected then.

File: api/templates/military-selective.xml (hunk 1 above)
Comment (gidjin) [reply api/templates/military-selective.xml]: Ah I think I follow the code now. Sorry for the confusion it was on my end.

**Pull request merged**